packages/backend/src/database/migrations/20230714124708_user_attributes.ts

```diff
@@ -0,0 +1,58 @@
+import { Knex } from 'knex';
+
+const userAttributesTableName = 'user_attributes';
+const organizationMemberUserAttributes = 'organization_member_user_attributes';
+
+export async function up(knex: Knex): Promise<void> {
+    if (!(await knex.schema.hasTable(userAttributesTableName))) {
+        await knex.schema.createTable(userAttributesTableName, (table) => {
+            table
+                .uuid('user_attribute_uuid')
+                .primary()
+                .defaultTo(knex.raw('uuid_generate_v4()'));
+            table.string('name').notNullable();
+            table.string('description').nullable();
+
+            table
+                .timestamp('created_at', { useTz: false })
+                .notNullable()
+                .defaultTo(knex.fn.now());
+            table
+                .integer('organization_id')
+                .notNullable()
+                .references('organization_id')
+                .inTable('organizations')
+                .onDelete('CASCADE');
+            table.unique(['name', 'organization_id']);
+            table.unique(['user_attribute_uuid', 'organization_id']); // needed for foreign key
```

Comment:
Is this just so it builds an index? You can use `.index` to make it obvious why it's there

Reply:
No, we require an unique constraint in order to use ` table .foreign(['user_attribute_uuid', 'organization_id'])` on `organizationMemberUserAttributes` 

An index is not enough

> alter table "organization_member_user_attributes" add constraint "organization_member_user_attributes_user_attribute_uuid_organization_id_foreign" foreign key ("user_attribute_uuid", "organization_id") references "user_attributes" ("user_attribute_uuid", "organization_id") on delete CASCADE - there is no unique constraint matching given keys for referenced table "user_attributes"
error: alter table "organization_member_user_attributes" add constraint "organization_member_user_attributes_user_attribute_uuid_organization_id_foreign" foreign key ("user_attribute_uuid", "organization_id") references "user_attributes" ("user_attribute_uuid", "organization_id") on delete CASCADE - there is no unique constraint matching given keys for referenced table "user_attributes"